handler/dataprep/create.go

```diff
@@ -77,6 +78,24 @@ func ValidateCreateRequest(ctx context.Context, db *gorm.DB, request CreateReque
 		return nil, errors.Wrap(handlererror.ErrInvalidParameter, "maxSize needs to be reduced to leave space for padding")
 	}
 
+	minPieceSizeStr := request.MinPieceSizeStr
+	if minPieceSizeStr == "" {
+		minPieceSizeStr = "1MiB"
+	}
+
+	minPieceSize, err := humanize.ParseBytes(minPieceSizeStr)
+	if err != nil {
+		return nil, errors.Join(handlererror.ErrInvalidParameter, errors.Wrapf(err, "invalid value for minPieceSize: %s", minPieceSizeStr))
+	}
+
+	if minPieceSize > pieceSize {
+		return nil, errors.Wrap(handlererror.ErrInvalidParameter, "minPieceSize cannot be larger than pieceSize")
+	}
+
+	if minPieceSize != util.NextPowerOfTwo(minPieceSize) {
+		return nil, errors.Wrap(handlererror.ErrInvalidParameter, "minPieceSize must be a power of two")
+	}
```

Comment:
It would be nice to add a check here to ensure the min-piece-size never over-pads in a way that makes things non-compliant.